solr/core/src/test/org/apache/solr/cloud/LeaderElectionTest.java

```diff
@@ -531,32 +533,30 @@ public void run() {
         };
 
     Thread connLossThread =
-        new Thread() {
-          @Override
-          public void run() {
-
-            while (!stopStress) {
-              try {
-                Thread.sleep(50);
-                int j;
-                j = random().nextInt(threads.size());
+        new Thread(
+            () -> {
+              while (!stopStress) {
                 try {
-                  threads.get(j).es.zkClient.getSolrZooKeeper().closeCnxn();
-                  if (random().nextBoolean()) {
-                    long sessionId = zkClient.getSolrZooKeeper().getSessionId();
-                    server.expire(sessionId);
+                  Thread.sleep(50);
+                  int j;
+                  j = random().nextInt(threads.size());
+                  try {
+                    ZooKeeper zk = threads.get(j).es.zkClient.getZooKeeper();
+                    assertTrue(zk instanceof TestableZooKeeper);
+                    ((TestableZooKeeper) zk).testableConnloss();
+                    if (random().nextBoolean()) {
+                      server.expire(zk.getSessionId());
```

Comment:
question: do we know/can we assume that `threads.get(j).es.zkClient` and `zkClient` use the same session id?

Reply:
I don't think they would, and I suspect this was a bug before. `zkClient` is the client for the text fixture itself, there should be no reason to be expiring the session on that one.

Reply:
Ahhh sorry I forgot to mention this Mike, I had already fixed it on my curator branch when we were talking.